components/elm/src/biogeophys/SedYieldMod.F90

```diff
@@ -128,71 +158,130 @@ subroutine SoilErosion (bounds, num_soilc, filter_soilc, &
             ! check landunit type and ground covered by snow/ice
             if ( lun_pp%itype(l)/=istsoil .and. lun_pp%itype(l)/=istcrop ) then
                cycle
-            else if (hsnow(c)>0) then
-               cycle
             end if
 
-            ! soil detachment by rainfall
+            ! check the glacier landunit in the gridcell
+            lg = grc_pp%landunit_indices(istice, g) 
+            fglacier = 1._r8
+            if (lg /= ispval) then
+               found = .false.
+               cg = lun_pp%coli(lg)
+               do while (.not. found .and. cg <= lun_pp%colf(lg))
+                  if (col_pp%active(cg)) then
+                     found = .true.
+                  else
+                     cg = cg + 1
+                  end if
+               end do
+               if (found .and. qflx_qrgwl(cg)<1e10_r8) then
```

Comment:
Add a decimal `1e10_r8` to `1.e10_r8`

Reply:
@tanzeli1982 I have made some changes related to the topounit structure.

Reply:
Thanks a lot.

Reply:
@tktesfa2014 Let me know if you have completed the review. I will then make all required updates.